### PR TITLE
ETFE 2355 - Add navigation banner to every page in the service

### PR DESCRIPTION
**Pull request by machinesleet**

### CODE REVIEW

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
The spacing between `Home` and `< Back` is quite small and could be difficult to interact for users with dexterity issues, it may be worth increasing the margin between these elements?
image

Also the navbar section doesn't change when the user clicks 'Movements', for example:
https://github.com/hmrc/emcs-tfe-frontend/assets/53828896/bfff582b-7a1a-4470-9a20-f375be61a036

File: app/controllers/predicates/AuthActionHelper.scala [hunk 1]
```diff
@@ -31,7 +31,7 @@ trait AuthActionHelper {
   private def authorised(ern: String): ActionBuilder[UserRequest, AnyContent] =
     auth(ern)
 
-  private def authorisedWithData(ern: String): ActionBuilder[DataRequest, AnyContent] =
+  def authorisedWithData(ern: String): ActionBuilder[DataRequest, AnyContent] =
```
[the remark this answers is not in the document]
Comment (machinesleet) [reply app/controllers/predicates/AuthActionHelper.scala:34]: When I removed the getMessageStatistics call in the AccountHomeController, the call to authorisedDataRequestAsync required the block to return a Future[Result]. I ended up having to wrap the view in a Future.successful to satisfy that type, whereas authorisedWithData didn't require this. Happy to change if you have any suggesstions.

**Comment by machinesleet** [comment]
@LewisMitchell  It looks like the spacing of the Back button matches the prototype. Is it worth asking design? 
I'll look in to the navbar section issue

File: app/views/components/navigation_bar.scala.html [hunk 2]
```diff
@@ -33,23 +32,23 @@
 
                 <ul class="moj-primary-navigation__list">
                     <li class="moj-primary-navigation__item">
-                        <a id="navigation-home-link" class="moj-primary-navigation__link" aria-current="page" href="@appConfig.emcsTfeHomeUrl">@messages("navigation.home")</a>
+                        <a id="navigation-home-link" class="moj-primary-navigation__link" @if(pageTitle == messages("viewMovement.title")) { aria-current="page" } href="@appConfig.emcsTfeHomeUrl">@messages("navigation.home")</a>
```
Comment (LewisMitchell) [inline app/views/components/navigation_bar.scala.html:35]: Matching on page title can be risky, each page will have a different title. IMO I think having some kind of viewmodel object that we pass to this component would be better. Like so:
https://github.com/hmrc/emcs-tfe-frontend/commit/f43725b4a1c2276c1c4c7d88604c69ffbce9a7ca

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -8,4 +8,8 @@ $hmrc-assets-path: "/emcs/account/assets/lib/hmrc-frontend/hmrc/assets/";
 
 .white-space-nowrap {
   white-space: nowrap;
+}
+
+.emcs-tfe-frontend-back-link {
+    margin-top: 30px;
```
Comment (MJCallahanPage) [inline app/assets/stylesheets/application.scss:14]: Probably better to use the existing `!important` CSS classes that you can use for alignment within the GovUk stylesheets.

https://design-system.service.gov.uk/styles/spacing/#overriding-spacing

This is a better option as the spacing resizes dynamically based on the screen size which is important for smaller devices such as mobiles.

File: app/views/templates/Layout.scala.html [hunk 4]
```diff
@@ -79,9 +82,10 @@
 
 @beforeContent = {
     @{ activeTrader(maybeShowActiveTrader) }
+    @maybeShowNavigationBanner.map(info => navigationBar(info))
     @beforeContentBlock.getOrElse(())
     @if(showBackLink) {
-        @govukBackLink(BackLinkViewModel(href = "#").withCssClass("js-visible"))
+        @govukBackLink(BackLinkViewModel(href = "#").withCssClass("js-visible").withCssClass("emcs-tfe-frontend-back-link"))
```
Comment (MJCallahanPage) [inline app/views/templates/Layout.scala.html:88]: Switch this to something like: `govuk-!-margin-top-6`

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
One small tweak to remove the custom CSS

**Review by james-work-account — DISMISSED** [review]

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Review by james-work-account — APPROVED** [review]

**Comment by james-work-account** [comment]
retest

**Pull request merged**